sdks/builds/unzip-android-archive.sh

```diff
@@ -15,14 +15,31 @@ trap cleanup 0
 #    $1: path to source archive
 #    $2: destination directory
 
-# This weird syntax is necessary because some zip archives do not contain a separate
-# entry for the root directory but merely a collection of its subdirectories (for instance
-# platform-tools). The very first entry in the archive is retrieved, then its path is read and
-# finally the part up to the first '/' of the path is retrieved as the root directory of
-# the archive. With platform-tools the last part is necessary because otherwise the root directory
-# would end up being reported as `platform-tools/adb` as this is the first entry in the archive and
-# there's no `platform-tools/` entry
-ZIP_ROOT_DIR=$(unzip -qql "$1" | head -n1 | tr -s ' ' | cut -d' ' -f5- | tr '/' ' ' | cut -d' ' -f1)
+#
+# Special case archives - archives which have no single root directory when unpacked and thus
+#                         the workaround below doesn't work for them.
+# Entry format: grep regex matching the END of path in $1 (no trailing $ necessary)
+#
+SPECIAL_CASE_ARCHIVES="cmake-.*\.zip"
```

Comment:
For "future maintenance" purposes, I would suggest doing:

```sh
SPECIAL_CASE_ARCHIVES="
    cmake-.*\.zip
    "
```

This will make it "obvious" where to put new entries.